File: backend/utils/instruments.py
"""Vocabulário fechado de instrumentos — compartilhado entre o perfil do
usuário (quais instrumentos ele toca, ver auth_service.py::user_instruments)
e "instrumentos buscados" de um anúncio do mural (band_board_service.py).
Precisa ser fechado (não texto livre) pra alertas por instrumento
(alerts_service.py) poderem comparar array contra array, sem depender de
correspondência aproximada de texto.

Não confundir com frontend/src/utils/tunings.js::INSTRUMENTS — aquela lista
é específica do Afinador (só 5 instrumentos com corda/afinação relevante),
esta é a lista geral de "instrumentos que um músico toca"."""
from __future__ import annotations

INSTRUMENTS = (
    "guitar", "bass", "drums", "vocals", "keys", "piano",
    "saxophone", "violin", "ukulele", "percussion", "other",
)

# mesma escala já usada em band_posts.skill_level (nível da banda como um
# todo) — reaproveitada aqui pro nível técnico POR instrumento do usuário.
SKILL_LEVELS = ("iniciante", "intermediario", "avancado", "profissional")

File: frontend/src/utils/instruments.js
// Vocabulário fechado de instrumentos — espelha backend/utils/instruments.py
// (mesmos ids dos dois lados). Usado no perfil do usuário (quais
// instrumentos ele toca) e nos "instrumentos buscados" de um anúncio do
// mural — precisa ser o mesmo vocabulário nos dois lugares pra alertas por
// instrumento funcionarem (comparação de array, não de texto aproximado).
//
// Não confundir com ./tunings.js::INSTRUMENTS (só 5 instrumentos, específico
// do Afinador/tuner, formato {id, labelKey, tunings}).
export const INSTRUMENTS = [
  'guitar', 'bass', 'drums', 'vocals', 'keys', 'piano',
  'saxophone', 'violin', 'ukulele', 'percussion', 'other',
]

// mesma escala já usada em bandBoard.json::form.skillLevels (nível da banda
// como um todo) — cópia própria pro nível técnico POR instrumento do
// usuário (ver Fase 1 do plano: mantidas independentes de propósito).
export const SKILL_LEVELS = ['iniciante', 'intermediario', 'avancado', 'profissional']
